include/sundials/sundials_math.h

```diff
@@ -75,83 +75,91 @@ extern "C" {
  */
 
 #ifndef SUNRsqrt
-#if defined(SUNDIALS_USE_GENERIC_MATH)
-#define SUNRsqrt(x) ((x) <= RCONST(0.0) ? (RCONST(0.0)) : ((realtype) sqrt((double) (x))))
-#elif defined(SUNDIALS_DOUBLE_PRECISION)
-#define SUNRsqrt(x) ((x) <= RCONST(0.0) ? (RCONST(0.0)) : (sqrt((x))))
-#elif defined(SUNDIALS_SINGLE_PRECISION)
-#define SUNRsqrt(x) ((x) <= RCONST(0.0) ? (RCONST(0.0)) : (sqrtf((x))))
-#elif defined(SUNDIALS_EXTENDED_PRECISION)
-#define SUNRsqrt(x) ((x) <= RCONST(0.0) ? (RCONST(0.0)) : (sqrtl((x))))
+#if (SUNDIALS_STDC_VERSION >= 199901L)
```

Comment:
I suppose so, but how else is it suppose to behave? Calling `sqrt` instead of `sqrtf` is still calling a std-c function in the sense we would still be following the API that the C standard defines. I suppose what is really important about `USE_GENERIC_MATH` is the second sentence in the docs:

> Results in linking to SUNDIALS_MATH_LIBRARY. 

Maybe we should just drop the first sentence which says

> Use the std-c math functions.

Reply:
I've been thinking of `USE_GENERIC_MATH` as a way for the user to indicate if `f` and `l` variants (C99 or compiler extensions) of math functions are available rather than relying on the C standard macros. Although the old behavior of this option was confusing:

* `USE_GENERIC_MATH = ON` -> link to `libm` and only call `double` math functions
* `USE_GENERIC_MATH = OFF` -> do not link to `libm` and use precision-specific math functions

The above behavior didn't make sense to me. Maybe it would be clearer to have a `SUNDIALS_C90_MATH` option/macro and if that is defined or true then only C90 math functions will be used (no `f` and `l` variants). The user could still control the math library with `SUNDIALS_MATH_LIBRARY` if necessary.